drivers/pinctrl/Kconfig.imx

```diff
@@ -6,3 +6,10 @@ config PINCTRL_IMX
 	depends on HAS_MCUX_IOMUXC || HAS_IMX_IOMUXC
 	help
 	  Enable pin controller driver for NXP iMX series MCUs
+
+# TODO: Find better place for this option
```

Comment:
For the benefit of other reviewers, the overview of the XBARA module from the RT1060 reference manual:

> This module implements an array of M N-input combinational muxes. All muxes share
the same N inputs in the same order, but each mux has its own independent select field.
The intended application of this module is to provide a flexible crossbar switch function
that allows any input (typically from external GPIO or internal module outputs) to be
connected to any output (typically to external GPIO or internal module inputs) under user
control. This is used to allow user configuration of data paths between internal modules
and between internal modules and GPIO.

This module is likely closest to a pin control driver, hence why I suggested to @bbilas to place the KConfig option here for the time being.